### PR TITLE
doc: add recommendation to use in-progress PR's

### PR DESCRIPTION
**Pull request by jasnell**
Per https://github.com/jasnell/dev-policy/issues/36, add a
recommendation to follow an "enhancement proposal" type of
process by using PR's with the `in-progress` tag. (looking
at https://github.com/iojs/io.js/pull/1159 as an example).

### CODE REVIEW

**Comment by Fishrock123** [comment]
Another example: https://github.com/iojs/io.js/pull/1457

File: README.md [hunk 1]
```diff
@@ -135,6 +135,10 @@ Additionally, Collaborators should:
 * Double check Pull Requests to make sure the author's full name and email address are correct before merging.
 * Verify that every commit passes all tests.
 
+### Seeking early review for significant changes
+
+When Collaborators propose to either make significant modifications to existing functionality or contribute new APIs and capabilities, the best practice is to follow an open "early review" process that allows other Collaborators and Node.js users to experiment with and provide feedback on the proposed contributions well before they land in `master`. One method of allowing this is to open a Pull Request with the *in-progress* tag that clearly indicates that the Pull Request is not yet ready to land. Such Pull Requests should contain enough detailed information about the changes being proposed to allow a reviewer to evaluate both the intent and the design of the changes.
```
Comment (Fishrock123) [inline README.md:140]: > One method of allowing this is to open a Pull Request with the _in-progress_ tag that clearly indicates that the Pull Request is not yet ready to land.

I don't really understand this entire "we need a tag to signify that this isn't ready to land".

if you're going to be merging it, you'd better understand it and be up to date with it anyways. :)

Also, non-collaborators cannot tag things themselves.

Comment (jasnell) [inline README.md:140]: It's really just a sign post. Some way of signaling that additional in-progress review is being requested. For the most part, however, it likely comes down to personal work mode choice.

Comment (mikeal) [inline README.md:140]: I've appreciated the `in-progress` label. I can't entirely describe why I find it so useful but it definitely helps my mental model of what the purpose of the PR is at any time.

**Pull request closed without merging**